tests/modules/test_text_reminder.py

```diff
@@ -469,3 +470,67 @@ def test_remind_when_good_dont_remind_when_cancelled(self, r_mocked_send_text, t
+        self.assertFalse(tr.correct_date_for_reminder(months_after_birth=16, days_before_appointment=1))
+        self.assertFalse(tr.correct_date_for_reminder(years_after_birth=5, days_before_appointment=7))
+        self.assertFalse(tr.correct_date_for_reminder(years_after_birth=5, days_before_appointment=1))
+        self.assertEqual(tr.get_reminder_msg(),
+                         verify_pregnant_signup_birthdate("English").format(name="Roland"))
+
+    @freeze_time(FAKE_NOW)
+    def test_remind_at_four_weeks_english(self):
+        tr = text_reminder_object("19/6/2017", preg_signup=True, preg_update=False) # 4 weeks, 0 days ago
+        self.assertFalse(tr.correct_date_for_reminder(weeks_after_birth=2, days_before_appointment=0))
+        self.assertTrue(tr.correct_date_for_reminder(weeks_after_birth=4, days_before_appointment=0))
+        self.assertTrue(tr.should_remind_today())
+        self.assertFalse(tr.correct_date_for_reminder(weeks_after_birth=6, days_before_appointment=7))
+        self.assertFalse(tr.correct_date_for_reminder(weeks_after_birth=6, days_before_appointment=1))
+        self.assertFalse(tr.correct_date_for_reminder(weeks_after_birth=10, days_before_appointment=7))
+        self.assertFalse(tr.correct_date_for_reminder(months_after_birth=9, days_before_appointment=7))
+        self.assertFalse(tr.correct_date_for_reminder(months_after_birth=9, days_before_appointment=1))
+        self.assertFalse(tr.correct_date_for_reminder(months_after_birth=16, days_before_appointment=7))
+        self.assertFalse(tr.correct_date_for_reminder(months_after_birth=16, days_before_appointment=1))
+        self.assertFalse(tr.correct_date_for_reminder(years_after_birth=5, days_before_appointment=7))
+        self.assertFalse(tr.correct_date_for_reminder(years_after_birth=5, days_before_appointment=1))
+        self.assertEqual(tr.get_reminder_msg(),
+                         verify_pregnant_signup_birthdate("English").format(name="Roland"))
+
+
+    @freeze_time(FAKE_NOW)
+    def test_remind_at_two_weeks_hindi(self):
+        tr = text_reminder_object("03/7/2017", language="Hindi", preg_signup=True, preg_update=False)
+        self.assertTrue(tr.correct_date_for_reminder(weeks_after_birth=2, days_before_appointment=0))
+        self.assertTrue(tr.should_remind_today())
+        self.assertEqual(tr.get_reminder_msg(),
+                         verify_pregnant_signup_birthdate('Hindi').format(name=u'\u0906\u0930\u0935'))
+
+    @freeze_time(FAKE_NOW)
+    def test_remind_at_four_weeks_hindi(self):
+        tr = text_reminder_object("19/6/2017", language="Hindi", preg_signup=True, preg_update=False)
+        self.assertTrue(tr.correct_date_for_reminder(weeks_after_birth=4, days_before_appointment=0))
+        self.assertTrue(tr.should_remind_today())
+        self.assertEqual(tr.get_reminder_msg(),
+                         verify_pregnant_signup_birthdate('Hindi').format(name=u'\u0906\u0930\u0935'))
```

Comment:
We should add tests that people _without_ pregnancy _do not_ get pregnancy updates.